src/runner.jl

```diff
@@ -87,77 +125,95 @@ function gettestfilepath(ctx::Context, pkgspec::Types.PackageSpec)
+    checkexitcode!(errs, proc, pkg, logfilename)
 
-**Keyword arguments:**
-  - `coverage::Bool=false`: enable or disable generation of coverage statistics.
-  - `julia_args::Union{Cmd, Vector{String}}`: options to be passed to the test process.
-  - `test_args::Union{Cmd, Vector{String}}`: test arguments (`ARGS`) available in the test process.
-  - `logfilepath::AbstractString=pwd()`: file path where test reports are saved.
-  - `logfilename::Union{AbstractString, Vector{AbstractString}}`: name(s) of test report file(s).
+Checks `proc.exitcode` and acts as follows:
 
-Generates a JUnit XML for the tests of package `pkg`, or for the current project
-(which thus needs to be a package) if no positional argument is given to
-`TestReports.test`. The test report is saved in the current working directory and
-called `testlog.xml` if both `logfilepath` and `logfilename` are not supplied.
-If `pkg` is of type `Vector{String}`, the report filenames are prepended with the
-package name, for example `Example_testlog.xml`.
+ - If 0, displays tests passed info message
+ - If equal to `TESTS_FAILED` const, warning is displayed and `pkg` added to `errs`
+ - If anything else, throws a `PkgTestError`
+"""
+function checkexitcode!(errs, proc, pkg, logfilename)
+    if proc.exitcode == 0
+        @info "$pkg tests passed. Results saved to $logfilename."
+    elseif proc.exitcode == TESTS_FAILED
+        @warn "ERROR: Test(s) failed or had an error in $pkg"
+        push!(errs, pkg)
+    else
+        throw(PkgTestError("TestReports failed to generate the report.\nSee error log above."))
+    end
+end
 
-If `logfilename` is supplied, it must match the type (and length, if a vector) of `pkg`.
+"""
+    runtests!(errs::Vector, pkg, cmd, logfilename)
 
-The tests are run in the same way as `Pkg.test`.
+Runs `cmd` which will run the unit tests of `pkg`. The exit code of the process
+is then checked.
 """
-function test(; kwargs...)
-    ctx = Context()
-    # This error mirrors the message generated by Pkg.test in similar situations
-    ctx.env.pkg === nothing && throw(PkgTestError("trying to test an unnamed project"))
-    test(ctx.env.pkg.name; kwargs...)
+function runtests!(errs::Vector, pkg, cmd, logfilename)
+    @info "Testing $pkg"
+    proc = open(cmd, Base.stdout; write=true)
+    wait(proc)
+    checkexitcode!(errs, proc, pkg, logfilename)
 end
-test(pkg::AbstractString; logfilename::AbstractString="testlog.xml", kwargs...) = test([pkg]; logfilename=[logfilename], kwargs...)
 
+"""
+    test!(pkg::AbstractString,
+          errs::Vector{AbstractString},
+          nopkgs::Vector{AbstractString},
+          notests::Vector{AbstractString},
+          logfilename::AbstractString;
+          coverage::Bool=false,
+          julia_args::Union{Cmd, AbstractVector{<:AbstractString}}=``,
+          test_args::Union{Cmd, AbstractVector{<:AbstractString}}=``)
+
+Tests `pkg` and save report to `logfilename`. Unit tests are run in the same way
```

Comment:
```suggestion
Tests `pkg` and save report to `logfilename`. Tests are run in the same way
```

They might be unit tests they might not..doesn't matter to us

Reply:
Good point, think I've said unit elsewhere too so will go and check